### ISSUE TITLE
Uninstall ATA Installed Types Packages if they Are No Longer in the TypesRegistry

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#15164]
**Repo**

1. @chrmarti  has previously used `require('vue')` in a js file and ATA had installed `@types/vue` on their machine.
2. He then recently created a new js file that uses `require('vue')`

**Expected behavior:**
The `@types/vue` package is no longer included in the `types-registry` since typings are now included in Vue itself. The old types should be removed from the ATA cache

**Actual behavior:**
The old `@types/vue` typings from the cache were used

**Comment by mhegazy** [comment]
Uninstall is nit the issue, module resolution should not include the old package.

Can you verify that node_modules/vue is available?

**Comment by chrmarti** [comment]
Sorry, not sure anymore which of my explorations this was with. I see both have `node_modules/vue`, but only one of these has typings configured in its `package.json`.

**Comment by mhegazy** [comment]
which one?

**Comment by chrmarti** [comment]
The vue package without typings is version 1.0.28. The one with is 2.2.6.

~/Library/Caches/typescript/node_modules/@types/vue/index.d.ts says "Type definitions for vuejs 1.0.21". (In case this helps.)

**Comment by mhegazy** [comment]
which one you have locally in node_modules? @1.0.28 or @2.2.6?

**Comment by chrmarti** [comment]
Tried to recreate the example without success, all the cases we thought problematic now work (TypeScript 2.3.4).

We found another bug, not sure if this is the same we originally saw and maybe misinterpreted. node_modules has vue 2.3.3.

Using:
```js
// @ts-check
import Vue from 'vue';

/** @type {Vue.Compo} */
var opt = {
     
};
```

In the comment we get suggestions from the global namespace instead of Vue:
![screen shot 2017-06-07 at 1 59 36 pm](https://user-images.githubusercontent.com/9205389/26901296-2424b4ee-4b8a-11e7-84dc-ace937fc0743.png)
